python/ray/ml/preprocessors/encoder.py

```diff
@@ -0,0 +1,129 @@
+from typing import List, Dict
+
+import pandas as pd
+
+from ray.data import Dataset
+from ray.data.aggregate import UniqueValues
+from ray.ml.preprocessor import Preprocessor
+
+
+class OrdinalEncoder(Preprocessor):
```

Comment:
A sanity check for all encoders should be to ensure there are no null values

Reply:
Added validation and updated tests accordingly (`fit`, `transform`, and `transform_batch` will all raise `ValueError` if a null value is detected).